test/parallel/test-http2-client-socket-operations.js

```diff
@@ -0,0 +1,99 @@
+'use strict';
+
+const common = require('../common');
+if (!common.hasCrypto)
+  common.skip('missing crypto');
+const h2 = require('http2');
+
+// test client.socket.destroy()
+{
+  const server = h2.createServer();
+  server.on(
+    'stream',
+    common.mustCall((stream) => {
+      // The stream aborted event must have been triggered
+      stream.on('aborted', common.mustCall());
+      stream.respond({
+        'content-type': 'text/html',
+        ':status': 200
+      });
+      stream.write('Hello World');
+    })
+  );
+
+  server.listen(
+    0,
+    common.mustCall(() => {
+      const client = h2.connect(`http://localhost:${server.address().port}`);
+      const req = client.request({ ':path': '/' });
+
+      req.on(
+        'response',
+        common.mustCall(() => {
+          // send a premature socket close
+          client.socket.destroy();
+        })
+      );
+      req.on('data', common.mustNotCall());
+
+      req.on(
+        'end',
+        common.mustCall(() => {
+          server.close();
+        })
+      );
+
+      // On the client, the close event must call
+      client.on('close', common.mustCall());
+      req.end();
+    })
+  );
+}
+
+// test pause, resume, drain
+{
+  const server = h2.createServer();
+  server.on(
+    'stream',
+    common.mustCall((stream) => {
```

Comment:
Ideally this would be on one line with `server.on('stream', common.mustCall((stream) => {`. See my general note re: using `prettier`.

Reply:
Moved it to one line in new commit